homeassistant/components/stream/core.py

```diff
@@ -356,3 +358,53 @@ async def handle(
     ) -> web.StreamResponse:
         """Handle the stream request."""
         raise NotImplementedError()
+
+
+class KeyFrameConverter:
+    """Generate and hold the keyframe as a jpeg."""
```

Comment:
A high level description of the threading model and state could be helpful here. I think its:
- the worker thread sets a packet
- the stream/home assistant loop can convert the packet to an image.
- the stream/home assistant loop has at most one worker at a time
- it will reuse an existing processed image, unless there is a new unprocessed packet

Reply:
Yes this is right, I can put that in.